AppController/djinn.rb

```diff
@@ -1834,27 +1837,32 @@ def job_start(secret)
       # restore_appcontroller_state modifies them.
       old_options = @options.clone
       old_jobs = my_node.jobs
+      my_versions_loaded = @versions_loaded if my_node.is_load_balancer?
 
-      # The following is the core of the duty cycle: start new apps,
-      # restart apps, terminate non-responsive AppServers, and autoscale.
+      # The appcontroller state is a misnomer, since it contains the state
+      # of the deployment that each node needs to reload. For example it
+      # contains the current listing of AppServers and Nodes.
+      unless restore_appcontroller_state
+        # Master is responsible to populate the state for the other nodes
+        # consumption, and since we know we could talk to zookeeper,
+        # master needs to go ahead and write the state.
+        unless my_node.is_shadow?
+          @state = "Couldn't reach the deployment state: now in isolated mode"
+          Djinn.log_warn("Cannot talk to zookeeper: in isolated mode.")
+          next
```

Comment:
Should there be a sleep here? Otherwise, this could be a busy loop for a disconnected AC.

Reply:
The busy loop should actually be in pick_zookeeper: I seem to remember that it was intentional, since an isolated node shouldn't quite be functional, so it is sitting waiting to reconnect to zookeeper. We could add a sleep in pick_zookeeper?

Reply:
Okay, I see now that the pick_zookeeper at least sleeps until the zk port is available, so I think this is fine.